### ISSUE TITLE
Restyle NetLogo dictionary

### ISSUE THREAD

**Issue by mrerrormessage** [NetLogo/NetLogo#911]
The NetLogo dictionary should look like it was written for a modern API/programming language. If anyone is interested in restyling it, I would welcome a PR. Some examples of visually attractive documentation:
- [Apple's Cocoa / iOS documentation](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/NSApplicationDelegate_Protocol/)
- [Android reference](http://developer.android.com/reference/android/app/Activity.html)
- [Google Apps Script documentation](https://developers.google.com/apps-script/reference/calendar/)

Scaladoc and Javadoc (for java 7 and 8) are also nice sources of inspiration

**Comment by SethTisue** [comment]
fwiw, Scaladoc received a facelift recently, coming in 2.12.0-M4: https://dl.dropboxusercontent.com/u/358427/scaladoc/library/index.html#package

**Comment by mrerrormessage** [comment]
Looks very nice! Even something on par with current (2.11) scaladoc would be a big visual upgrade from where we're at right now.

**Comment by BruceEdmonds** [comment]
No, no no!  I think you will find that the current style NetLogo Dictionary is well loved.  The above "professional" looking versions are nowhere near as user-friendly.

**Comment by mrerrormessage** [comment]
I completely agree that we want to keep the NetLogo Dictionary user-friendly. The above examples I provided feature very complicated APIs and I certainly don't want to complicate the NetLogo documentation needlessly. I propose to *enhance* the ease-of-use of the dictionary. Some ideas:

* Use color to indicate whether a given entry is a keyword, command, reporter, variable, or constant
* Avoid overly-long lists of related primitives by breaking them into separate sections where necessary
* Provide examples where needed, but avoid listing primitives twice to no effect, see screenshots:

<img width="155" alt="screen shot 2016-01-14 at 2 31 28 pm" src="https://cloud.githubusercontent.com/assets/143198/12336715/935fbb16-bacb-11e5-97d6-d905d0921578.png">

<img width="461" alt="screen shot 2016-01-14 at 2 20 23 pm" src="https://cloud.githubusercontent.com/assets/143198/12336688/6a6cfd86-bacb-11e5-819f-ea46374cc0f3.png">

Changing the dictionary will necessarily be an adjustment for long-time users, but hopefully we can make that transition worthwhile.

**Comment by Idloj** [comment]
Another source of inspiration might be https://www.blender.org/manual/

**Comment by reyman** [comment]
I'm agree with @BruceEdmonds, Netlogo Dictionnary is really good in this actual format.

Perhaps something like asciidoctor language/tool (http://asciidoctor.org/ which also exist in java http://asciidoctor.org/docs/asciidoctorj/) could produce great documentation output in different format easily (html, pdf, epub,etc.), and don't really differ it seems from the actual way that netlogo build the doc. See the keyword/primite [include](http://asciidoctor.org/docs/user-manual/#include-directive) of this langage for example.

**Comment by nicolaspayette** [comment]
Somewhat related: https://github.com/NetLogo/NetLogo/issues/616.

**Comment by ERSUCC** [comment]
The docs have undergone a significant visual overhaul for NetLogo 7, you can check them out [here](https://docs.netlogo.org/7.0.0-beta2/)!
